hswidget/ds1widget/helper.go

```diff
@@ -0,0 +1,49 @@
+package ds1widget
+
+import (
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
+)
+
+// Warning: this is 1:1 copy from
+// github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2ds1.(*DS1).setupLayerStreamType()
+// but this method is unexported for now, so...
+// see https://github.com/OpenDiablo2/OpenDiablo2/pull/1059
+func (p *DS1Widget) recreateLayerStreamTypes() {
```

Comment:
1059 has been closed; this is a bad idea to put state management code for `d2ds1` outside of `d2ds1`, and an even worse idea to make users of the package responsible for knowing when to call state management/upkeep methods.

we need to take a closer look at `d2ds1` and resolve state management there before we merge this PR.